src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,63 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def get_slack_actions_with_async_lookups(organization, user, data):
+    try:
+        from sentry.incidents.endpoints.serializers import AlertRuleTriggerActionSerializer
+
+        slack_actions = []
+        for trigger in data["triggers"]:
+            for action in trigger["actions"]:
+                action = rewrite_trigger_action_fields(action)
+                a_s = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "access": SystemAccess(),
+                        "user": user,
+                    },
+                    data=action,
+                )
+                if a_s.is_valid():
+                    if (
+                        a_s.validated_data["type"].value == AlertRuleTriggerAction.Type.SLACK.value
+                        and not a_s.validated_data["input_channel_id"]
+                    ):
+                        slack_actions.append(a_s.validated_data)
+    except KeyError:
```

Comment:
@iProgramStuff BTW, where can this key error happen? It might be better to put the `try/except` only around the part that can generate a `KeyError`. It also makes it clear for future people looking at the code where the `KeyError` might happen which is generally helpful. Also, a comment why the `KeyError` might happen would be helpful.

Reply:
It can happen anywhere we're accessing the lists if they don't have the key. I think it could even happen when accessing validated_data (I'd have to verify).

Reply:
@iProgramStuff What were your findings about `KeyError`? Is it only happening when we do `a_s.validated_data["input_channel_id"]`? If so, seems like we should just do `a_s.validated_data.get("input_channel_id")`

Reply:
I think it's best to leave it as-is. It could also happen when accessing triggers and actions, since that is raw data submitted from the api.

Reply:
@iProgramStuff makes sense, could you add that as a comment?